tools/lkl/lib/nt-host.c

```diff
@@ -88,25 +89,25 @@ static int thread_join(lkl_thread_t tid)
 	return 0;
 }
 
-static int tls_alloc(unsigned int *key)
+static int tls_alloc(unsigned int *key, void (*destructor)(void *))
 {
-	*key = TlsAlloc();
+	*key = FlsAlloc((PFLS_CALLBACK_FUNCTION)destructor);
 	return *key == TLS_OUT_OF_INDEXES ? -1 : 0;
 }
 
 static int tls_free(unsigned int key)
 {
-	return TlsFree(key) ? 0 : -1;
+	return FlsFree(key) ? 0 : -1;
```

Comment:
I did it like this to prevent the callback from being called when one calls `tls_free`

     static int tls_free(unsigned int key)
     {
    -       return TlsFree(key) ? 0 : -1;
    +       // Setting to NULL first to prevent the callback from being called
    +       if (!FlsSetValue(key, NULL))
    +    {
    +        return -1;
    +    }
    +       return FlsFree(key) ? 0 : -1;
     }